receiver/googlecloudspannerreceiver/internal/metadata/metricsdatapoint.go

```diff
@@ -59,3 +80,45 @@ func (mdp *MetricsDataPoint) GroupingKey() MetricsDataPointKey {
 		MetricDataType: mdp.metricValue.DataType(),
 	}
 }
+
+func (mdp *MetricsDataPoint) ToItem() (*filter.Item, error) {
+	seriesKey, err := mdp.hash()
+	if err != nil {
+		return nil, err
+	}
+
+	return &filter.Item{
+		SeriesKey: seriesKey,
+		Timestamp: mdp.timestamp,
+	}, nil
+}
+
+func (mdp *MetricsDataPoint) toDataForHashing() dataForHashing {
+	// Do not use map here because it has unpredicted order
+	// Taking into account 3 default labels: project_id, instance_id, database
+	labels := make([]label, len(mdp.labelValues)+3)
+
+	labels[0] = label{Name: projectIDLabelName, Value: mdp.databaseID.ProjectID()}
+	labels[1] = label{Name: instanceIDLabelName, Value: mdp.databaseID.InstanceID()}
+	labels[2] = label{Name: databaseLabelName, Value: mdp.databaseID.DatabaseName()}
+
+	labelsIndex := 3
+	for _, labelValue := range mdp.labelValues {
+		labels[labelsIndex] = label{Name: labelValue.Name(), Value: labelValue.Value()}
+		labelsIndex++
+	}
+
+	return dataForHashing{
+		MetricName: mdp.metricName,
+		Labels:     labels,
+	}
+}
+
+func (mdp *MetricsDataPoint) hash() (string, error) {
+	hashedData, err := hashstructure.Hash(mdp.toDataForHashing(), nil)
```

Comment:
We are using `fnv` elsewhere in the codebase. Can you use it as well and avoid this external dependency?

Reply:
I used external library because it can hash entire structure - that was the intent. Will look on your proposal if it can work in same way today.

Reply:
Ok, I checked the `fnv` and it doesn't feet my needs, because it works with byte arrays and `hashstructure` can hash entire structure(you can find this in MR), uses `fnv` under the hood, has a lot of useful options, handle zero values and more: https://github.com/mitchellh/hashstructure (in other words - it handles a lot of boilerplate and manual things).